misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -94,20 +96,24 @@ internal class SqsJobConsumer @Inject internal constructor(
               val (duration, _) = timed { handler.handleJob(message) }
               metrics.handlerDispatchTime.record(duration.toMillis().toDouble(), queueName.value)
             } catch (th: Throwable) {
-              failure.set(true)
               log.error(th) { "error handling job from ${queueName.value}" }
               metrics.handlerFailures.labels(queueName.value).inc()
               Tags.ERROR.set(span, true)
+              throw th
             }
           }
         }
       }
 
-      return when {
-        failure.get() -> Status.FAILED
-        messages.size == 0 -> Status.NO_WORK
-        else -> Status.OK
+      for (future in futures) {
+        try {
+          future.get()
+        } catch (e: Exception) {
```

Comment:
not `Throwable`?

Reply:
it doesn't really matter here, you're not trying to prevent an Exception from being leaked (RepeatedTaskQueue handles that). Just trying to catch the signal that the dispatched handler threw an exception.